native/src/main/java/io/ballerina/lib/milvus/Utils.java

```diff
@@ -79,4 +91,91 @@ private static Object convertToJsonFields(Object value) {
         }
         return value;
     }
+
+    protected static void generateSearchResult(RecordType outputFieldsType, List<SearchResp.SearchResult> results,
+                                             BMap<BString, Object>[] responses) {
+        for (SearchResp.SearchResult result : results) {
```

Comment:
Shall we remove the `responses` array parameter and return a new array from the method? This will eliminate side effects and make the code easier to reason about.

Reply:
Fixed in [9592a2f](https://github.com/ballerina-platform/module-ballerinax-milvus/pull/7/commits/9592a2f4b36aa23d2888c9f135982ca78bdb1b77)